### DISCUSSION TITLE
How to store the pieces of information for the secrets?

### DISCUSSION THREAD

**Discussion by giorgionocera** [nabla-studio/nablajs#1] [Q&A]
Since we need to manage multiple `mnemonic seed phrases`, we need to interact with different values, which are encrypted to increase the security.

For this reason, we need to store at least this information:
- the **passphraseHash**, which is the same for all the mnemonics (this reduce the security but increases the UX);
- the **currentMnemonicIndex**, which corresponds to the index of the mnemonic the user selected to use;
- the **cipherMetadata** object, that contains information about the cipher method (e.g., the **cipherType**, the **keyLength**, the **keyGenerationMethod**, etc.)
- an array with n **mnemonics** objects, where each should contains:
  - the **name**;
  - the **cipherText**, which is the cipher mnemonic;
  - the **cipheredMetadata**, corresponding to information linked to encryption/decryption activities (e.g., the **IV** for an AES method encryption, parameters for the key generation method).

In such a scenario, I think these pieces of information can be stored in the async storage as 4 different objects:
- `passphraseHash`, string as _plainText_
- `currentMnemonicIndex`, integer as _plainText_
- `cipherMetadata`, object as _plainText_
- `mnemonics`, which is the array of objects (_plaintext_ overall)
  - `name`, string as _plainText_
  - `cipherText`, string as _cipherText_
  - `cipheredMetadata`, object as _plainText_
  
What do you think about this schema?

**Comment by DavideSegullo** [comment]
I think this scheme is fine, if you agree we can continue the development

**Reply by giorgionocera** [reply]
Okay, let's move!
